src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -264,36 +266,7 @@ int IDebugPendingBreakpoint2.Bind()
                             }
                         }
                     }
-
-                    Task bindTask = null;
-                    _engine.DebuggedProcess.WorkerThread.RunOperation(() =>
-                    {
-                        bindTask = _engine.DebuggedProcess.AddInternalBreakAction(this.BindAsync);
-                    });
-
-                    bindTask.Wait(_engine.GetBPLongBindTimeout());
-                    if (!bindTask.IsCompleted)
-                    {
-                        //send a low severity warning bp. This will allow the UI to respond quickly, and if the mi debugger doesn't end up binding, this warning will get 
-                        //replaced by the real mi debugger error text
-                        this.SetError(new AD7ErrorBreakpoint(this, ResourceStrings.LongBind, enum_BP_ERROR_TYPE.BPET_SEV_LOW | enum_BP_ERROR_TYPE.BPET_TYPE_WARNING), true);
-                        return Constants.S_FALSE;
-                    }
-                    else
-                    {
-                        if ((enum_BP_LOCATION_TYPE)_bpRequestInfo.bpLocation.bpLocationType == enum_BP_LOCATION_TYPE.BPLT_DATA_STRING)
-                        {
-                            lock (_engine.DebuggedProcess.DataBreakpointVariables)
-                            {
-                                string addressName = HostMarshal.GetDataBreakpointStringForIntPtr(_bpRequestInfo.bpLocation.unionmember3);
-                                if (!_engine.DebuggedProcess.DataBreakpointVariables.Contains(addressName)) // might need to expand condition
-                                {
-                                    _engine.DebuggedProcess.DataBreakpointVariables.Add(addressName);
-                                }
-                            }
-                        }
-                        return Constants.S_OK;
-                    }
+                    return BindWithTimeout();
```

Comment:
>return [](start = 20, length = 6)

I think we want: `if (!_enabled) { return S_OK; }`

Reply:
Yes, you're right. Otherwise we're requesting a pointless disabled breakpoint from GDB. With this change, I don't need to do the somewhat inelegant "always unbind on enable" to deal with initially disabled breakpoints, [so I've removed it](https://github.com/microsoft/MIEngine/pull/1128/commits/5514dd680be7dd755d3ef35e6fec48bd4e1f9452#diff-ceef2a8399bc4087d455776c21f6948685c24b4967202803d844289332003856L486-L500).